packages/kit/src/runtime/client/renderer.js

```diff
@@ -275,6 +275,8 @@ export class Renderer {
 			this._init(navigation_result);
 		}
 
+		getSelection()?.removeAllRanges();
+
```

Comment:
not exactly related to your change, but `if (!opts)` shouldn't the `document.body.focus()` line still be called? I don't quite get what that check is doing and wonder if it should be removed? maybe i'm just sleep deprived enough that I'm making a dumb mistake in reading this code...

Reply:
I'm actually quite confused of the part inside `if (!opts)` too. When I refactored that, I tried to keep it as before. That means `document.body.focus()` and scrolling only happens if `opts` exist, I thought it was intentional.

If it wasn't, maybe we should refactor that again, and always do focusing and scrolling regardless of `opts`. We can do that in a separate PR though.